File: test_data/code/python/1299-replace-elements-with-greatest-element-on-right-side.py
Python
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        n = len(arr)
        i = n - 1
        max_to_right = -1
        
        while i >= 0:
            temp = arr[i]
            arr[i] = max_to_right
            if i < n - 1:
                max_to_right = max(max_to_right, arr[i + 1])
            i -= 1
        
        return arr

File: test_data/code/java/1299-replace-elements-with-greatest-element-on-right-side.java
class Solution {
    public int[] replaceElements(int[] arr) {
        int n = arr.length;
        if (n == 0) {
            return new int[0];
        }
        
        int max = arr[n - 1];
        arr[n - 1] = -1;
        
        for (int i = n - 2; i >= 0; i--) {
            if (arr[i] > max) {
                max = arr[i];
            }
            arr[i] = max;
        }
        
        return arr;
    }
}